kitsune/wiki/views.py

```diff
@@ -1445,3 +1473,116 @@ def what_links_here(request, document_slug):
     }
 
     return render(request, 'wiki/what_links_here.html', c)
+
+
+def get_fallback_locale(doc, request):
+    """Get best fallback local based on locale Mapping"""
+
+    # Get the Translated locales which have current revision
+    translated_locales = (doc.translations.all()
+                                          .values_list('locale', flat=True)
+                                          .exclude(current_revision=None))
+    # Get the Accepted_language header from user request
+    browser_language_code = request.META.get('HTTP_ACCEPT_LANGUAGE', None)
+    if browser_language_code is not None:
+        header_locales = parse_accept_lang_header(browser_language_code)
+    else:
+        header_locales = []
+    # Get the Locale mapping from setting
+    NON_SUPPORTED_LOCALES = settings.NON_SUPPORTED_LOCALES
+
+    # Check if the document is translated to any of the locale listed in Accepted_Language Header
+    for locale in header_locales:
+        if locale in translated_locales:
+            fallback_locale = locale
+            break
+    else:
+        # If any locale of Accpeted_Language Header is not in translated locale,
+        # Check if we have any fallback local for the locales of Accpeted_Languge Header and
+        # the document is translated into that fallback locale
+        # This locale mapping is mentioned in kitsune/settings.py
+        for locale in header_locales:
+            if NON_SUPPORTED_LOCALES.get(locale) in translated_locales:
+                fallback_locale = NON_SUPPORTED_LOCALES[locale]
+                break
+        else:
+            # If above does not match, then check if we have custom Wiki Locale mapping
+            # for the user requested locale.
+            # This custom Wiki Locale Mapping is mentioned in kitsune/wiki/config.py
+            if request.LANGUAGE_CODE in FALLBACK_LOCALES:
+                for locale in FALLBACK_LOCALES[request.LANGUAGE_CODE]:
+                    if locale in translated_locales:
+                        fallback_locale = locale
+                        break
+                else:
+                    # If above does not match, then we would check if any locale of
```

Comment:
Entire section rewrite:
# If the above did not match, check for matches of any of the ACCEPT_LANGUAGE header locales in the custom wiki locale mapping